quarkchain/cluster/shard_state.py

```diff
@@ -806,14 +806,20 @@ def __rewrite_block_index_to(self, minor_block, add_tx_back_to_queue=True):
             self.db.remove_minor_block_index(block)
             if add_tx_back_to_queue:
                 self.__add_transactions_from_block(block)
+        if len(old_chain) > 0:
             asyncio.ensure_future(
-                self.subscription_manager.notify_log(block.header, True)
+                self.subscription_manager.notify_log(block.header, len(old_chain), True)
             )
         for block in new_chain:
             self.db.put_transaction_index_from_block(block)
             self.db.put_minor_block_index(block)
             self.__remove_transactions_from_block(block)
-            asyncio.ensure_future(self.subscription_manager.notify_log(block.header))
+        # new_chain has at least one block, starting from minor_block with block height descending
+        new_chain.sort(key=lambda x: x.header.height)
+        asyncio.ensure_future(self.subscription_manager.notify_new_heads(new_chain))
+        asyncio.ensure_future(
+            self.subscription_manager.notify_log(minor_block.header, len(new_chain))
```

Comment:
so `minor_block == new_chain[0]`?

Reply:
seems so? the only place `new_chain` gets appended is line 797 and `block` starts from `minor_block`.